lib/services/vector_store.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import os
+from typing import List, Optional
+
+from langchain_openai import OpenAIEmbeddings
+from langchain_postgres import PGVector
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from pydantic import BaseModel, Field
+from sqlalchemy.ext.asyncio import create_async_engine
+
+from lib.config.env import config
+
+logger = logging.getLogger(__name__)
+
+# RAG retrieval settings
+RAG_TOP_K = 30  # Number of passages to retrieve per query
+
+# RAG chunking settings (all in characters)
+RAG_CHUNK_SIZE = 2000  # Characters per chunk when indexing documents
+RAG_CHUNK_OVERLAP = 400  # Character overlap between adjacent chunks
+
+EMBEDDING_MODEL = "text-embedding-3-large"
+
+
+class RetrievedPassage(BaseModel):
+    """Represents a passage retrieved from vector store."""
+
+    content: str = Field(description="The text content of the retrieved passage")
+    source_file: str = Field(description="Name of the source file")
+    chunk_index: int = Field(description="Index of the chunk within the source")
+    similarity_score: float = Field(description="Cosine similarity score (0-1)")
+    page_number: Optional[str] = Field(
+        default=None, description="Page number if available"
+    )
+
+
+def get_file_hash_from_path(file_path: str) -> str:
+    """Extract xxh128 hash from file path where hash is the filename."""
+    return os.path.basename(file_path)
+
+
+def get_collection_id(file_hash: str) -> str:
+    """Generate pgvector collection ID from file hash."""
+    return f"doc_passages_{file_hash[:16]}"
+
+
+class VectorStoreService:
+    """Service for vector storage and retrieval operations."""
+
+    def __init__(self, connection_string: str):
+        """Initialize vector store with database connection."""
+        self.embeddings = OpenAIEmbeddings(model=EMBEDDING_MODEL)
+        self.chunker = RecursiveCharacterTextSplitter(
+            chunk_size=RAG_CHUNK_SIZE,
+            chunk_overlap=RAG_CHUNK_OVERLAP,
+            separators=["\n\n", "\n", ". ", " ", ""],
+        )
+
+        # Always use async engine since all our methods are async
+        # Convert postgresql:// to postgresql+psycopg:// for SQLAlchemy async engine
+        if connection_string.startswith("postgresql://"):
+            async_url = connection_string.replace(
+                "postgresql://", "postgresql+psycopg://", 1
```

Comment:
something I reminded just now: I believe you'll need to add the name of the tables created by langchain's rag to the `IGNORE_TABLES` array defined in `alembic/env.py`, otherwise the next alembic migration will try to delete these tables